Scripts/Runtime/Entities/Lifecycle/AbstractEntityLifecycleStatusSystem.cs

```diff
@@ -0,0 +1,303 @@
+using Anvil.CSharp.Collections;
+using Anvil.Unity.DOTS.Jobs;
+using System.Collections.Generic;
+using Unity.Burst;
+using Unity.Collections;
+using Unity.Entities;
+using Unity.Jobs;
+using UnityEngine;
+
+namespace Anvil.Unity.DOTS.Entities
+{
+    /// <summary>
+    /// Provides information on the lifecycle of entities.
+    /// </summary>
+    /// <remarks>
+    /// Best practice is to extend this system and place the newly created system at a point in time when structural
+    /// changes have been made. Ex. Right after a <see cref="EntityCommandBufferSystem"/>.
+    /// Created or Imported entities will be treated as "Arrivals" to the World.
+    /// Destroyed or Evicted entities will be treated as "Departures" from the World.
+    /// Entities that are in a "CleanUp" state can be treated as "Departures" before they are actually destroyed by
```

Comment:
This appears to be an option not a default or hard-coded behaviour.
I'm happy to have this as the default behaviour but we should specify that and set the default on the constructor param.

Reply:
Ok sure, updated.